libnavigation-ui/src/main/java/com/mapbox/navigation/ui/route/MapRouteProgressChangeListener.kt

```diff
@@ -66,14 +68,23 @@ internal class MapRouteProgressChangeListener(
                 job = ThreadController.getMainScopeAndRootJob().scope.launch {
                     val totalDist =
                         (routeProgress.distanceRemaining() + routeProgress.distanceTraveled())
-                    val dist = routeProgress.distanceTraveled() / totalDist
-                    if (dist > 0) {
-                        val deferredExpression = async(Dispatchers.Default) {
-                            routeLine.getExpressionAtOffset(dist)
+                    val percentDistanceTraveled = routeProgress.distanceTraveled() / totalDist
```

Comment:
IMO `updateRoute` is getting too long. It'd be great if we could simplify it extracting blocks of code into `private` methods. You know, we should keep methods as clean as possible so they read like pseudocode (i.e. no conditionals, `for`s, etc.).

Reply:
Good point. I'll clean it up.